### PR TITLE
Added Perf Package

### PR DESCRIPTION
**Pull request by colbyhall**
The idea of this package is for measuring performance. To start off the package I've added the high res query counter functionality with procedures for time measuring.

### CODE REVIEW

**Comment by oskarnp** [comment]
Unix version is not correct.  tv_nsec only contains the nanosecond part of the seconds. So need both tv_sec and tv_nsec.

File: core/perf/perf_windows.odin [hunk 1]
```diff
@@ -0,0 +1,30 @@
+// +build windows
+package perf
+
+import "core:sys/win32"
+
+@(private="file")
+qpc_freq : i64;
```
Comment (Tetralux) [inline core/perf/perf_windows.odin:7]: Might want to consider making this `@thread_local`.

**Comment by colbyhall** [comment]
So whats the next step to get this merged?

**Comment by gingerBill** [comment]
I have now integrated this code into commit 9e8c46b8, and reorganized it by placing it in `package time`

**Pull request closed without merging**